nav2_lifecycle_manager/src/lifecycle_manager.cpp

```diff
@@ -295,6 +317,7 @@ bool
 LifecycleManager::shutdown()
 {
   system_active_ = false;
```

Comment:
You can remove `system_active_` and replace the times its checked with `state_of_managed_nodes_ == NodeState::ACTIVE` as an inlined separate method `isActive()`.

Reply:
Good point, change made!